docs/source/frequency_guide.rst

```diff
@@ -0,0 +1,212 @@
+User Guide for Frequency Control
+================================
+
+Description
+-----------
+GEOPM supports many interfaces to manipulate frequency. This guide provides
+specifics on each of these interfaces including suggested use, implementation
+nuances, and common pitfalls. Supported interfaces include:
+
+* Direct frequency requests on CPU core/uncore and GPU
+* Hardware P-States (HWP) on CPU
+* Speed Select Technology - Core Priority (SST-CP) and Turbo Frequency (SST-TF)
+  on Intel CPUs ICX+
+
+GEOPM frequency decisions interact with the OS frequency drivers. This
+interaction is detailed below. At a high level, frequency drivers may include
+settings to drive frequency with the objective of achieving greater performance
+or to lower power consumption, while other frequency driver settings may adhere
+more strictly to user frequency requests. The interaction between driver and
+GEOPM-driven frequency setting will vary depending upon the interface used.
+Generally speaking, if a ``userspace`` governor is available on a given driver,
+it is more likely to play nicely with GEOPM.
+
+The ideal mechanism of frequency control depends of course on the use-case and
+priorities of the user. If performance repeatability is critical or the user
+knows of ideal frequency settings, direct frequency requests are likely ideal.
+If the user wishes to leverage as much of the power headroom as possible, SST
+or HWP interfaces are useful. If the user also wishes the system to steer power
+intelligently between cores/uncore based on internal telemetry, HWP will work
+best. If computation on a given node is heterogeneous (i.e. some CPUs are given
+more work or more critical work than others), SST features are likely to work
+best.
+
+Linux Frequency Drivers
+-----------------------
+
+Linux CPU driver information can be found using ``cpupower`` as shown in the
+example below. If the driver has a ``userspace`` governor available, that will
```

Comment:
Minor suggestion just to provide a little bit more context for someone not familiar with Linux's scaling governors:
```suggestion
example below. If the driver has a ``userspace`` CPU governor available, that will
```

More instances below. Feel free to ignore.